### ISSUE TITLE
Internal Linking not working

### ISSUE THREAD

**Issue by jwynn6** [asciidoctor/asciidoctor-epub3#150]
I have a document with lots of internal linking, and linking between included sub-documents. Asciidoctor-pdf generates the document correctly, but asciidoctor-epub3 fails to generate a single link properly. 

asciidoctor: WARNING: <tite>.adoc: invalid reference to unknown local anchor (or valid bibref): <link text>

**Comment by jnavila** [comment]
Some Id generations were missing in the latest beta7. Try compiling and installing the lastest master from this repo.

BTW @mojavelinux Do you think a new beta would be possible?

**Comment by mojavelinux** [comment]
Yes, that is the plan. I just need to review this issue and one other, then I'll move forward with the release.

**Comment by agoncal** [comment]
@mojavelinux  I am running into the same problem with Asciidoctor 2.0. Internal links work ok in PDF but not in EPUB. Any work around ?

**Comment by mojavelinux** [comment]
I realize this is tricky. At this point the only workaround is to ensure the processor produces an interdocument reference in EPUB3 and a local reference for PDF. That's done by:

* Giving the target a globally unique ID
* Giving the chapter an ID that matches the filename
* Putting the chapter files in the root folder
* Including the chapter into the master document (aka spine) using an include
* Always using an interdocument xref to refer to the sections in other chapters

In other words, the reference should be written as:

```
<<chapter-1.adoc#unique-section-id,text>>
```

or

<<chapter-1.adoc#chapter-1,Chapter 1>>

(Notice that the ID of the chapter has to be used even though you are referring to the chapter file itself).

This really needs to be documented in the README clearly because it's quite difficult to get right.

**Comment by mojavelinux** [comment]
(If there is still a problem with the .adoc file extension for some reason, just drop it).

**Comment by mojavelinux** [comment]
Btw, beta.8 is out now.

**Comment by agoncal** [comment]
Hum... This is what I have

```
spine.adoc
  include::jpa-chapter-mapping.adoc[leveloffset=+1]
  include::jpa-chapter-advanced.adoc[leveloffset=+1]

jpa-chapter-managing.adoc
  [[jpa-listing-managing-1]]

jpa-chapter-advanced.adoc
  <<jpa-listing-managing-1>> this is the reference to the chapter mapping listing
```

And if I add the adoc reference as you suggest, it doesn't work either:

```
spine.adoc
  include::jpa-chapter-mapping.adoc[leveloffset=+1]
  include::jpa-chapter-advanced.adoc[leveloffset=+1]

jpa-chapter-managing.adoc
  [[jpa-listing-managing-1]]

jpa-chapter-advanced.adoc
  <<jpa-chapter-managing.adoc#jpa-listing-managing-1>>
```

Any idea ?

**Comment by slonopotamus** [comment]
I declare this issue to be a duplicate of #210.
